### PR TITLE
Revise `PixelImageBlock` to correctly use the "new" `next/image` component

### PR DESCRIPTION
**Pull request by SebiVPS**
Revise `PixelImageBlock` to use the updated next image component (remove layout prop)

- remove layout prop from `PixelImageBlock`
- remove fixed and unused disableBlurPlaceholder
- remove unnecessary `NextImageBottomPaddingFix`

Use the `fill` prop to fill in the div with the given or inherited (calculated from the file) aspect ratio.
If one sets fill manually, one can embed the image in its own container.

### CODE REVIEW

File: packages/site/cms-site/src/blocks/PixelImageBlock.tsx [hunk 1]
```diff
@@ -11,14 +12,10 @@ import { PropsWithData } from "./PropsWithData";
 
 interface PixelImageBlockProps extends PropsWithData<PixelImageBlockData>, Omit<ImageProps, "src" | "width" | "height" | "alt"> {
     aspectRatio: string | "inherit";
```
Comment (SebiVPS) [inline packages/site/cms-site/src/blocks/PixelImageBlock.tsx:14]: @johnnyomair @mariokemetinger should we change this "inherit" string to something else? (eg. "file") 
If this is set, the aspectRatio from the file is calculated. 
Inherit would mean, that the dimensions from the parent should be used.
The function can be called `computeAspectRatioFromCrop`

**Comment by nsams** [comment]
With this, when updating to v7 (and next 13) devs are forced to update all images.

There is no easier-to-update alternative (as next provides with their legacy image)

Is this acceptable?

**Comment by SebiVPS** [comment]
> With this, when updating to v7 (and next 13) devs are forced to update all images.
> There is no easier-to-update alternative (as next provides with their legacy image)
> Is this acceptable?

In most cases the adaptions like in this PR `demo/site/src/blocks/DamImageBlock.tsx` should be enough i think. (just deletions!). This would be acceptable.

I asked @mariokemetinger yesterday, if we can try to evaluate some special cases, where more adaptions must be done.

**Comment by johnnyomair** [comment]
Could we provide `LegacyPixelImageBlock` and `LegacyImage` to ease migration?

File: packages/site/cms-site/src/blocks/PixelImageBlock.tsx [hunk 2]
```diff
@@ -67,34 +52,34 @@ export const PixelImageBlock = withPreview(
...
-                placeholder="blur"
-                blurDataURL={blurDataUrl}
-                alt={damFile.altText ?? ""}
-                {...nextImageProps}
-            />
+            <ImageContainer $aspectRatio={usedAspectRatio}>
+                <NextImage
+                    loader={(loaderProps) => generateImageUrl(loaderProps, usedAspectRatio)}
+                    src={urlTemplate}
+                    fill
```
Comment (johnnyomair) [inline packages/site/cms-site/src/blocks/PixelImageBlock.tsx:76]: Why do you have the special treatment of `fill` if you add it here as well?

File: packages/site/cms-site/src/blocks/PixelImageBlock.tsx [hunk 3]
```diff
@@ -111,3 +96,9 @@ function createDominantImageDataUrl(w: number, h: number, dominantColor = "#ffff
 
     return `data:image/svg+xml;base64,${toBase64(svg)}`;
 }
+
+const ImageContainer = styled.div<{ $aspectRatio: number }>`
+    position: relative;
+    width: 100%;
+    aspect-ratio: ${({ $aspectRatio }) => $aspectRatio};
+`;
```
Comment (johnnyomair) [inline packages/site/cms-site/src/blocks/PixelImageBlock.tsx:104]: Why is this needed? I can imagine that this leads to many unwanted behaviors

File: .changeset/friendly-mayflies-play.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-site": major
+---
+
+Revise `PixelImageBlock` to use updated next image component (remove layout prop)
```
Comment (johnnyomair) [inline .changeset/friendly-mayflies-play.md:5]: Changesets for breaking changes should be detailed. What do I need to do to upgrade? Maybe link to the Next upgrade guide (if it exists)

**Review by johnnyomair — COMMENTED** [review]
This needs also be in the migration guide

File: packages/site/cms-site/src/blocks/PixelImageBlock.tsx [hunk 5]
```diff
@@ -11,14 +12,10 @@ import { PropsWithData } from "./PropsWithData";
 
 interface PixelImageBlockProps extends PropsWithData<PixelImageBlockData>, Omit<ImageProps, "src" | "width" | "height" | "alt"> {
     aspectRatio: string | "inherit";
-    layout: "fill" | "fixed" | "intrinsic" | "responsive";
-    // Workaround to prevent flicker on already loaded images.
-    // See https://github.com/vercel/next.js/issues/27539 for more information.
-    disableBlurPlaceholder?: boolean;
```
Comment (SebiVPS) [inline packages/site/cms-site/src/blocks/PixelImageBlock.tsx:17]: removed this prop, it is not used in any of our projects and its fixed in next v11.2.1 (https://github.com/vercel/next.js/issues/27539#issuecomment-923618301)

File: .changeset/friendly-mayflies-play.md (hunk 4 above)
Comment (SebiVPS) [reply .changeset/friendly-mayflies-play.md]: i updated the changeset.

File: packages/site/cms-site/src/blocks/PixelImageBlock.tsx (hunk 3 above)
Comment (SebiVPS) [reply packages/site/cms-site/src/blocks/PixelImageBlock.tsx:95]: as discussed, the container is needed for our default behaviour

File: packages/site/cms-site/src/blocks/PixelImageBlock.tsx (hunk 2 above)
Comment (SebiVPS) [reply packages/site/cms-site/src/blocks/PixelImageBlock.tsx]: as discussed, for a customisation option i offer a version without container, when the fill prop is set

**Review by mariokemetinger — DISMISSED** [review]

**Review by nsams — APPROVED** [review]
Maybe also add ad link to https://nextjs.org/docs/pages/api-reference/components/image-legacy#comparison

(and if we need a Legacy image support, I can not commet)

**Review by johnnyomair — APPROVED** [review]

File: .changeset/friendly-mayflies-play.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+---
+"@comet/cms-site": major
+---
+
+Rework `PixelImageBlock` to correctly use the "new" `next/image` component
```
Comment (johnnyomair) [inline .changeset/friendly-mayflies-play.md:5]:
```suggestion
Revise `PixelImageBlock` to correctly use the "new" `next/image` component
```

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**